docs/4.x/tracking-java.md

```diff
@@ -0,0 +1,169 @@
+
+The tracker uses primarily the authentication set via method parameter, then the authentication set in the request, and finally the authentication set in the tracker configuration. If no authentication token is set, the tracker won't set the parameter `token_auth` in the tracking request. This can lead to problems if you use parameters that require an authentication token. In this case, the tracker will throw an exception on sending the request.
+
+Please mind that the token should be transmitted securely, for example using HTTPS. 
+
+## Custom Tracking Parameters
+
+You can also set additional custom request parameters using the method `MatomoRequest.request().additionalParameters(Map<String, Collection<Object>>)`. The additional parameters will be added to the request URL as query parameters. The query parameter name will be the key of the map entry and the query parameter value will be the value of the map entry. To add the same query parameter name with multiple values, you can use a collection as the parameter value, e.g. `additionalParameters(Map.of("foo", List.of("bar", "baz")))` will add the query parameters `?foo=bar&foo=baz` to the request URL (note that the order of the query parameters is not guaranteed). Note: If you set any parameter that is not supported by the Matomo tracking API, the parameter will be ignored by Matomo, although it will be added to the request URL.
+
+To see which parameters are available, have a look at the [JavaDoc of MatomoRequest](https://matomo-org.github.io/matomo-java-tracker/javadoc/org/matomo/java/tracking/MatomoRequest.html) and the [Matomo Tracking API Reference](https://developer.matomo.org/api-reference/tracking-api) for your Matomo version.
+
+## Ecommerce Items
+
+Matomo supports ecommerce tracking, i.e. tracking of ecommerce actions like adding items to the cart, removing items from the cart, and purchasing items. Every ecommerce action can be tracked as a goal. To track ecommerce actions, you need to set a goal id in the request. You can set the goal id using the method `MatomoRequest.request().goalId(int)`. The goal id is the id of the goal in Matomo. You can find the goal id in the Matomo user interface under "Administration" -> "Websites" -> "Goals". The goal id is the number in the URL of the edit page.
+
+Ecommerce items can be added to a request using the method `MatomoRequest.request().ecommerceItems(EcommerceItems)`. The items will be added to the request URL as a JSON representation in the query parameter `ec_items`. The JSON representation is an array of JSON objects. Each JSON object represents an ecommerce item. Ecommerce items are only supported if a goal id is set in the request. If you set ecommerce items without a goal id, the items will be ignored by Matomo.
```

Comment:
I actually don't understand the reason of requiring a goal id. For ecommerce actions the goal id `0` needs to be sent. Not sure what sending a goal id with another value might cause. So why isn't the goal id automatically set when a ecommerce action is tracked?

Reply:
That's a nice idea! I opened an issue here: https://github.com/matomo-org/matomo-java-tracker/issues/162